operate/common/src/main/java/io/camunda/operate/conditions/DatabaseInfo.java

```diff
@@ -22,18 +22,21 @@ public class DatabaseInfo implements ApplicationContextAware {
   static final DatabaseType DEFAULT_DATABASE = DatabaseType.Elasticsearch;
   private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseInfo.class);
   private static ApplicationContext applicationContext;
+  private static DatabaseType current;
 
-  public static DatabaseType getCurrent() {
-    if (applicationContext == null) {
-      LOGGER.warn("getCurrent() called on DatabaseInfo before application context has been set");
-      return DEFAULT_DATABASE;
+  public static synchronized DatabaseType getCurrent() {
```

Comment:
:thought_balloon: I don't think this has to be `synchronized`. Having it synchronized introduces the same performance problem again where all threads fight for this lock.